great_expectations/rule_based_profiler/config/base.py

```diff
@@ -0,0 +1,282 @@
+import copy
+import logging
+from typing import Any, Dict, List, Optional
+
+from great_expectations.data_context.types.base import BaseYamlConfig
+from great_expectations.marshmallow__shade import INCLUDE, Schema, fields, post_load
+from great_expectations.marshmallow__shade.decorators import post_dump
+from great_expectations.types import DictDot
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class DomainBuilderConfig(DictDot):
+    def __init__(
+        self,
+        class_name: str,
+        module_name: Optional[str] = None,
+        batch_request: Optional[Dict[str, Any]] = None,
+        **kwargs,
+    ):
+        self._class_name = class_name
+        self._module_name = module_name
+        self.batch_request = batch_request
+        for k, v in kwargs.items():
+            setattr(self, k, v)
```

Comment:
Not sure if this is 100% correct. Whenever we identify an unknown value, do we want to set it as an attr? I don't think so but I didn't want to possibly lose values.

Reply:
Keeping the values allows us to maintain state when writing back to disk.

Reply:
Seems fine to me...

Reply:
No longer saving attrs; resolving